### ISSUE TITLE
Unhandled Errors in Fetch Promises

### ISSUE THREAD

**Issue by wolfymaster** [cosmicjs/cosmicjs-node#13]
Curious why there is no catch on the fetches or any way to handle a reject. I think this is in most of the code, but I came across it looking at getObject. 

```
  fetch(endpoint)
    .then(function(response){
      if (response.status >= 400) {
        var err = {
          'message': 'There was an error with this request.'
        }
        return callback(err, false);
      }
      return response.json()
    })
    .then(function(response){
      // Constructor
      var cosmic = {};
      var object = response.object;
      var metafields = object.metafields;
      if(metafields){
        object.metafield = _.keyBy(metafields, "key");
      }
```

If the resources does not exist, Cosmic returns a 404, which passes an error message back to my callback, but then continues to the next then() statement and errors out at response.object. Is there a way that I am supposed to halt the fetch() in my callback to prevent it from executing the final then() when there is an error ?

**Comment by tonyspiro** [comment]
Thanks for opening the issue @wolfymaster!  Looks like there needs to be a throw on the callbacks and a catch on each fetch.  What do you think?  Would very much appreciate a PR if you have some time to fix this up.

**Comment by wolfymaster** [comment]
That was my line of thought. I can submit a PR for this. Thanks.

**Comment by tonyspiro** [comment]
Thank you @wolfymaster 👊

**Comment by tonyspiro** [comment]
Merged in v3.0
